content/posts/2021/test-python-functions/index.md

```diff
@@ -0,0 +1,387 @@
+---
+title: "Using Tox and PyTest with OpenFaaS"
+description: "How to Unit Test Python functions in OpenFaaS"
+date: 2021-03-06T00:00:00+02:00
+tags:
+  - programming
+  - openfaas
+  - serverless
+  - python
+  - pytest
+  - tox
+  - ci/cd
+  - github actions
+  - testing
+  - unit test
+draft: false
+# <span>Photo by <a href="https://unsplash.com/@alex_andrews?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Alexander Andrews</a> on <a href="https://unsplash.com/?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Unsplash</a></span>
+coverImage: images/cover2-alexander-andrews.jpg
+---
+
+I think it is an uncontroversial statement that to say testing is important in software development. Writing tests may not always be fun, but nothing is a sweet as that moment when a unit test catches a bug before your commit.
+
+In OpenFaaS we have tons of tests in each project, even the [`certifier`](https://github.com/openfaas/certifier) itself that runs a short suite of end-to-end tests. _But_, not all of our function templates have first class testing support. For Go, Node, and Java, this has already been added. For Go it is easy because there is a standard test command `go test`. For Node and Java we (or the community) have picked and optional default test command. Soon, we are going to do the same for Python.
```

Comment:
the second half of the first sentence makes sense to me.  Should it be `even the certifier itself runs a short suite of end-to-end tests`?